pxr/usd/usdShade/validators.cpp

```diff
@@ -192,13 +243,164 @@ _ShaderPropertyTypeConformance(const UsdPrim &usdPrim)
+                    name);
+            }
+        );
+    for (const UsdProperty& matBindingProperty : matBindingProperties) {
+        if (matBindingProperty.Is<UsdRelationship>()) {
+            ++numMatBindingRels;
+        }
+    }
+
+    if (numMatBindingRels < 1u) {
+        return {};
+    }
+
+    if (subset.GetFamilyNameAttr().HasAuthoredValue()) {
+        return {};
+    }
+
+    const UsdValidationErrorSites primErrorSites = {
+        UsdValidationErrorSite(usdPrim.GetStage(), usdPrim.GetPath())
+    };
+
+    return {
+        UsdValidationError(
+            UsdValidationErrorType::Error,
+            primErrorSites,
+            TfStringPrintf(
+                "GeomSubset prim <%s> with material bindings applied but "
+                "no authored family name should set familyName to '%s'.",
+                usdPrim.GetPath().GetText(),
+                UsdShadeTokens->materialBind.GetText())
+        )
+    };
+}
+
+static
+UsdValidationErrorVector
+_SubsetsMaterialBindFamily(const UsdPrim& usdPrim)
+{
+    if (!(usdPrim && usdPrim.IsInFamily<UsdGeomImageable>(
+            UsdSchemaRegistry::VersionPolicy::All))) {
+        return {};
+    }
+
+    const UsdGeomImageable imageable(usdPrim);
+    if (!imageable) {
+        return {};
+    }
+
+    const std::vector<UsdGeomSubset> materialBindSubsets =
+        UsdGeomSubset::GetGeomSubsets(
+            imageable,
+            /* elementType = */ TfToken(),
+            /* familyName = */ UsdShadeTokens->materialBind);
+
+    if (materialBindSubsets.empty()) {
+        return {};
+    }
+
+    UsdValidationErrorVector errors;
+
+    // Check to make sure that the "materialBind" family is of a restricted
+    // type, since it is invalid for an element of geometry to be bound to
+    // multiple materials.
+    const TfToken materialBindFamilyType = UsdGeomSubset::GetFamilyType(
+        imageable,
+        UsdShadeTokens->materialBind);
+    if (materialBindFamilyType == UsdGeomTokens->unrestricted) {
+        const UsdValidationErrorSites primErrorSites = {
+            UsdValidationErrorSite(usdPrim.GetStage(), usdPrim.GetPath())
+        };
+
+        errors.emplace_back(
+            UsdValidationErrorType::Error,
+            primErrorSites,
+            TfStringPrintf(
+                "Imageable prim <%s> has '%s' subset family with invalid "
+                "family type '%s'. Family type should be '%s' or '%s' "
+                "instead.",
+                usdPrim.GetPath().GetText(),
+                UsdShadeTokens->materialBind.GetText(),
+                materialBindFamilyType.GetText(),
+                UsdGeomTokens->nonOverlapping.GetText(),
+                UsdGeomTokens->partition.GetText())
+        );
+    }
+
+    // Check that all subsets belonging to the "materialBind" family are of
+    // element type "face", since material bindings may only be applied to
+    // geometric faces.
```

Comment:
Since usdPhysics [also uses](https://github.com/PixarAnimationStudios/OpenUSD/blob/release/pxr/usd/usdPhysics/overview.dox#L403-L408) the materialBindingAPI, we now have use cases for binding materials to non-face subsets! Since [createMaterialBindSubsets](https://github.com/PixarAnimationStudios/OpenUSD/blob/release/pxr/usd/usdShade/materialBindingAPI.h#L909-L927) allows any elementType as well, I don't think we need this validation check anymore.

If you agree with that @mattyjams, I can go ahead and update your PR internally before pushing it!

Reply:
Oh, interesting! Sure, that sounds good to me, thanks @diyajoy!